h/static/scripts/controllers/form-controller.js

```diff
@@ -0,0 +1,204 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var { findRefs, setElementState } = require('../util/dom');
+var submitForm = require('../util/submit-form');
+
+/**
+ * A controller which adds inline editing functionality to forms
+ */
+class FormController extends Controller {
+  constructor(element, options) {
+    super(element, options);
+
+    setElementState(this.refs.cancelBtn, {hidden: false});
+    this.refs.cancelBtn.addEventListener('click', event => {
+      event.preventDefault();
+      this.cancel();
+    });
+
+    // List of groups of controls that constitute each form field
+    this._fields = Array.from(element.querySelectorAll('.js-form-input'))
+      .map(el => {
+        var parts = findRefs(el);
+        return {container: el, input: parts.formInput};
+      });
+
+    this.on('focus', event => {
+      var field = this._fields.find(field => field.input === event.target);
```

Comment:
Heheh, the polyfill for IE/Edge is basically identical to my hastily written code above (except that `Element#closest` will match the current element as well:

https://github.com/jonathantneal/closest/blob/master/closest.js